packages/contracts/contracts/libs/v0.8.x/minter-libs/SplitFundsLib.sol

```diff
@@ -159,25 +190,27 @@ library SplitFundsLib {
                     _valueInWei,
                 "Invalid revenue split totals"
             );
-        }
-        // Render Provider / Art Blocks payment
-        if (renderProviderRevenue_ > 0) {
-            (success, ) = renderProviderAddress_.call{
-                value: renderProviderRevenue_
-            }("");
-            require(success, "Render Provider payment failed");
-        }
-        // artist payment
-        if (artistRevenue_ > 0) {
-            (success, ) = artistAddress_.call{value: artistRevenue_}("");
-            require(success, "Artist payment failed");
-        }
-        // additional payee payment
-        if (additionalPayeePrimaryRevenue_ > 0) {
-            (success, ) = additionalPayeePrimaryAddress_.call{
-                value: additionalPayeePrimaryRevenue_
-            }("");
-            require(success, "Additional Payee payment failed");
+            // Render Provider / Art Blocks payment
+            if (renderProviderRevenue_ > 0) {
+                (bool success, ) = renderProviderAddress_.call{
+                    value: renderProviderRevenue_
+                }("");
+                require(success, "Render Provider payment failed");
+            }
+            // artist payment
+            if (artistRevenue_ > 0) {
+                (bool success, ) = artistAddress_.call{value: artistRevenue_}(
+                    ""
+                );
+                require(success, "Artist payment failed");
+            }
+            // additional payee payment
+            if (additionalPayeePrimaryRevenue_ > 0) {
+                (bool success, ) = additionalPayeePrimaryAddress_.call{
+                    value: additionalPayeePrimaryRevenue_
+                }("");
+                require(success, "Additional Payee payment failed");
+            }
```

Comment:
super small nit for this function in general, other than sourcing the payments, much of this code is duplicate. would it make sense to refactor the shared parts into another func ( + conditional provider payment if engine === true)

Reply:
good call! I was battling stack too deep issues, but this was a smol disaster 😆 updated to use some helper functions and eliminate the redundant code in [792d0e6](https://github.com/ArtBlocks/artblocks-contracts/pull/1069/commits/792d0e684be26eebff2a394bdc7bf6750f91fc55)